### PR TITLE
prov/verbs: Bug fix for matching domain name with device name

### PR DESCRIPTION
**Pull request by j-xiong**
The domain name may contain a `-<xxx>` suffix, the original code used `strncmp(domain_name, dev_name, strlen(dev_name))` to check if the names match. However, this scheme does not work when the device name is a substring of the domain name (e.g. `mlx5_11-rdm` vs `mlx5_1`) and would incorrectly return match for such case.

### CODE REVIEW

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -1220,6 +1220,24 @@ vrb_info_add_dev_addr(struct fi_info **info, struct verbs_dev_info *dev)
 	return 0;
 }
 
+/* domain name may have a "-<xxx>" suffix */
+static inline int vrb_cmp_domain_and_dev_name(const char *domain_name,
+					      const char *dev_name)
+{
+	size_t cmp_len;
+	char *s = strchr(domain_name, '-');
```
Comment (shefty) [inline prov/verbs/src/verbs_info.c:1228]: Is there any chance '-' could appear in the device name?  If so, strrchr() may work.

**Review by shefty — APPROVED** [review]

File: prov/verbs/src/verbs_info.c (hunk 1 above)
Comment (shefty) [reply prov/verbs/src/verbs_info.c:1228]: Unless we could have 2 suffixes...

Comment (j-xiong) [reply prov/verbs/src/verbs_info.c:1228]: Theoretically yes, but practically no.  I can change it to `strrchr()` just in case.

Comment (j-xiong) [reply prov/verbs/src/verbs_info.c]: The suffix is controlled by the provider. We only have one.

**Pull request merged**